cli/cli_app.go

```diff
@@ -367,6 +368,12 @@ func assumeRoleIfNecessary(terragruntOptions *options.TerragruntOptions) error {
 	terragruntOptions.Env["AWS_SECRET_ACCESS_KEY"] = aws.StringValue(creds.SecretAccessKey)
 	terragruntOptions.Env["AWS_SESSION_TOKEN"] = aws.StringValue(creds.SessionToken)
 
+	// Workaround for https://github.com/hashicorp/go-getter/issues/191
+	// Set these directly as environment variables so that go-getter can download artifacts from S3
+	os.Setenv("AWS_ACCESS_KEY_ID", aws.StringValue(creds.AccessKeyId))
+	os.Setenv("AWS_SECRET_ACCESS_KEY", aws.StringValue(creds.SecretAccessKey))
+	os.Setenv("AWS_SESSION_TOKEN", aws.StringValue(creds.SessionToken))
```

Comment:
Thanks for the PR!

I'm grateful for the attempt at a simple fix, but I'm 95% sure this approach is going to cause other problems:

1. We run Terragrunt's tests in parallel, some of which use different auth strategies. If we set global env vars here, that will affect all tests, and cause strange, intermittent behavior depending on which tests happen to run in parallel.
1. Terragrunt itself runs things in parallel when you run `xxx-all` commands (e.g., `apply-all`). If you have different modules that use different auth settings, this again will cause strange race conditions based on timing.

So, I suspect that either we need a way to explicitly set auth values when executing an instance of `go-getter`, if it's API exposes that ability... And if not, fix the bug in `go-getter` itself.